integration/spark/app/src/test/java/io/openlineage/spark/agent/SparkGenericIntegrationTest.java

```diff
@@ -0,0 +1,120 @@
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.mockserver.configuration.Configuration;
+import org.mockserver.integration.ClientAndServer;
+import org.slf4j.event.Level;
+
+/**
+ * This class contains Spark non-container integration tests that do not fit into other integration
+ * test classes.
+ */
+@Tag("integration-test")
+@Slf4j
+public class SparkGenericIntegrationTest {
+
+  @SuppressWarnings("PMD")
+  private static final String LOCAL_IP = "127.0.0.1";
+
+  private static final int MOCKSERVER_PORT = 1081;
+
+  private static ClientAndServer mockServer;
+
+  static SparkSession spark;
+
+  @BeforeAll
+  @SneakyThrows
+  public static void beforeAll() {
+    SparkSession$.MODULE$.cleanupAnyExistingSession();
+    Configuration configuration = new Configuration();
+    configuration.logLevel(Level.ERROR);
+    mockServer = ClientAndServer.startClientAndServer(configuration, MOCKSERVER_PORT);
+    mockServer
+        .when(request("/api/v1/lineage"))
+        .respond(org.mockserver.model.HttpResponse.response().withStatusCode(201));
+  }
+
+  @AfterAll
+  @SneakyThrows
+  public static void afterAll() {
+    SparkSession$.MODULE$.cleanupAnyExistingSession();
+    mockServer.stop();
+  }
+
+  @BeforeEach
+  @SneakyThrows
+  public void beforeEach() {
+    spark =
+        SparkSession.builder()
+            .master("local[*]")
+            .appName("GenericIntegrationTest")
+            .config("spark.driver.host", LOCAL_IP)
+            .config("spark.driver.bindAddress", LOCAL_IP)
+            .config("spark.sql.shuffle.partitions", 1)
+            .config("spark.openlineage.transport.type", "http")
+            .config(
+                "spark.openlineage.transport.url",
+                "http://localhost:" + mockServer.getPort() + "/api/v1/namespaces/generic-namespace")
+            .config("spark.openlineage.facets.disabled", "spark_unknown;spark.logicalPlan")
+            .config("spark.openlineage.debugFacet", "disabled")
+            .config("spark.openlineage.parentJobName", "parent-job")
+            .config("spark.openlineage.parentRunId", "bd9c2467-3ed7-4fdc-85c2-41ebf5c73b40")
+            .config("spark.openlineage.parentJobNamespace", "parent-namespace")
+            .config("spark.extraListeners", OpenLineageSparkListener.class.getName())
+            .getOrCreate();
+
+    spark.sparkContext().setLogLevel("WARN");
+  }
+
+  @Test
+  public void sparkEmitsApplicationLevelEvents() {
+    Dataset<Row> df = createTempDataset();
+
+    Dataset<Row> agg = df.groupBy("a").count();
+    agg.write().mode("overwrite").csv("/tmp/test_data/test_output/");
+
+    spark.stop();
+    verifyEvents(
+        mockServer,
+        "applicationLevelStartApplication.json",
```

Comment:
Writing assertions here would allow verifying if a runId is shared among different events

Reply:
Done.